xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-watch/src/main/java/org/xwiki/notifications/filters/watch/WatchedEntityReference.java

```diff
@@ -42,17 +48,85 @@ enum WatchedStatus
         /**
          * The entity is watched for all events and all formats: the filter doesn't specify any event or format.
          */
-        WATCHED_FOR_ALL_EVENTS_AND_FORMATS,
+        WATCHED_FOR_ALL_EVENTS_AND_FORMATS(true, false),
+
+        /**
+         * The entity is watched for all events and all formats through a filter placed on an ancestor.
+         * @since 16.5.0RC1
+         */
+        @Unstable
+        WATCHED_BY_ANCESTOR_FOR_ALL_EVENTS_AND_FORMATS(true, false),
+
+        /**
+         * The entity and its children are watched for all events and all formats.
+         * @since 16.5.0RC1
+         */
+        @Unstable
+        WATCHED_WITH_CHILDREN_FOR_ALL_EVENTS_AND_FORMATS(true, false),
+
+        /**
+         * The entity does not have a watched status: by default it means it's not watched, but it's not blocked.
+         *
+         * @since 15.9RC1
+         */
+        NOT_SET(false, false),
```

Comment:
No unstable there?

Reply:
Hmm there's a wrong since on this one I have been fooled because of it I think.

Reply:
Bad copy/paste it seems I'll fix that.